components/encoder/ams/ams_as5048.go

```diff
@@ -151,28 +171,21 @@ func (enc *Encoder) Reconfigure(
 	if err != nil {
 		return err
 	}
-
-	brd, err := board.FromDependencies(deps, newConf.BoardName)
-	if err != nil {
-		return err
-	}
-	localBoard, ok := brd.(board.LocalBoard)
-	if !ok {
-		return errors.Errorf(
-			"board with name %s does not implement the LocalBoard interface", newConf.BoardName,
-		)
-	}
-	i2c, exists := localBoard.I2CByName(newConf.I2CBus)
-	if !exists {
-		return errors.Errorf("unable to find I2C bus: %s", newConf.I2CBus)
-	}
 	enc.mu.Lock()
 	defer enc.mu.Unlock()
-	if enc.i2cBus == i2c || enc.i2cAddr == byte(newConf.I2CAddr) {
-		return nil
+
+	if enc.i2cBusName != newConf.I2CBus || enc.i2cBus == nil {
```

Comment:
I grant that these are the conditions in which we _need_ to initialize a new bus, but is it ever harmful to do so? Like, what if we removed the if statement and re-initialized it every time? The code would be less complicated, and we wouldn't need to store the bus name any more.

I haven't thought this through: maybe there's a good reason to keep the old bus around if it hasn't changed. Keep the PR as-is if you want.